### PR TITLE
feat: add icons to button

### PR DESCRIPTION
**Pull request by Atzac**

### CODE REVIEW

File: src/components/button/button.test.tsx [hunk 1]
```diff
@@ -17,57 +17,91 @@ function sut(props: ButtonProps = defaultButton): HTMLElement {
...
-    }
-  );
+    });
+
+    it('should render icon in right side', () => {
+      const icon = 'alert';
+      expect(
+        sut({ ...defaultButton, icon: icon, iconOnRight: true }).className
+      ).toContain('iconOnRight-true');
+    });
```
Comment (iurynogueira) [inline src/components/button/button.test.tsx:105]: I suggest create a spec to check if o icon is on left when the attr iconOnRight not is informed

File: src/components/button/button.tsx [hunk 2]
```diff
@@ -13,25 +15,41 @@ export type ButtonProps = {
...
+  };
+
   return (
     <ButtonStyles
       type={type}
       onClick={handleClick}
       disabled={disabled}
       danger={danger}
       size={size}
+      withIcon={icon && true}
```
Comment (iurynogueira) [inline src/components/button/button.tsx:46]: this true is really necessary?

File: src/components/button/button.tsx [hunk 3]
```diff
@@ -13,25 +15,41 @@ export type ButtonProps = {
...
+      iconOnRight={iconOnRight}
     >
-      {label}
+      <div>
+        {icon && <IonIcon type={icon} size={iconSize[`${size}`]} />}
```
Comment (iurynogueira) [inline src/components/button/button.tsx:50]:
```suggestion
        {icon && <IonIcon type={icon} size={iconSize[`${size}`]} />}
```

https://itnext.io/7-bad-code-smells-in-typescript-e1cb397723c6

check topic 4.

**Review by iurynogueira — APPROVED** [review]

File: src/stories/button/button.stories.tsx [hunk 4]
```diff
@@ -55,3 +55,18 @@ PrimarySmall.args = {
   type: 'primary',
   size: 'sm',
 };
+
```
Comment (iurynogueira) [inline src/stories/button/button.stories.tsx:58]: ...

**Review by iurynogueira — CHANGES_REQUESTED** [review]

File: src/stories/button/button.stories.tsx (hunk 4 above)
Comment (iurynogueira) [reply src/stories/button/button.stories.tsx:58]: disregard it

File: src/components/button/button.tsx (hunk 3 above)
Comment (Atzac) [reply src/components/button/button.tsx:50]: https://pt-br.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

File: src/components/button/button.tsx (hunk 2 above)
Comment (Atzac) [reply src/components/button/button.tsx:46]: Yes. 'withIcon' is a variant that only activates if an icon exists. The property only accepts boolean. Do you have a better suggestion?

Comment (iurynogueira) [reply src/components/button/button.tsx:46]: !!icon

**Review by wermesonsl — APPROVED** [review]

File: src/components/button/button.test.tsx [hunk 5]
```diff
@@ -11,63 +11,108 @@ const defaultButton: ButtonProps = {
...
+
+    it('should render icon in right side', () => {
+      const icon = 'alert';
+      sut({ ...defaultButton, icon: icon, iconOnRight: true });
+      expect(getButton().className).toContain('iconOnRight-true');
+    });
+
+    it('should render icon in left by default', () => {
+      sut({ ...defaultButton, icon: 'alert', iconOnRight: false });
```
Comment (iurynogueira) [inline src/components/button/button.test.tsx:111]:
```suggestion
      sut({ ...defaultButton, icon: 'alert'});
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**